File: Models/PageModel/PageStyle/Style.cs
using System;
using System.Collections.Generic;

namespace Model.PageModel.PageStyle
{
    public class Style
    {
        public string Height { get; set; }
        public string Width { get; set; }
        public SpaceProperty Margin { get; set; }
        public string Background { get; set; }
        public string TextColor { get; set; } 
        public string FontFamily { get; set; }
        public string FontSize { get; set; }
        public SpaceProperty Padding { get; set; }
        public BorderProperty Borders { get; set; }
    }
}

File: resources/Models/style.ts
export interface Style {

    Width?: string;

    Height?: string;

    Margin?: SpaceProperty;

    Background?: string;

    TextColor?: string;

    FontFamily?: string;

    FontSize?: string;

    Padding?: SpaceProperty;

    Borders?: BorderProperty[];

}

export interface SpaceProperty {

    Total?: string;

    Top?: string;

    Bottom?: string;

    Left?: string;

    Right?: string;

}

export interface BorderProperty {

    Type: BorderType;

    Style: string;

    Width: string;

    Color: string;

}

export enum BorderType {

    Total = 0,

    Left = 1,

    Right = 2,

    Top = 3,
    
    Bottom = 4

}